Sources/client/Socket.swift

```diff
@@ -147,51 +150,37 @@ public class Socket {
   //----------------------------------------------------------------------
   // MARK: - Initialization
   //----------------------------------------------------------------------
+  public convenience init(_ endPoint: String) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: { nil })
+  }
+
   public convenience init(_ endPoint: String,
-                          params: [String: Any]? = nil) {
+                          params: Payload?) {
     self.init(endPoint: endPoint,
               transport: { url in return WebSocket(url: url) },
-              params: params)
+              params: { params })
+  }
+
+  public convenience init(_ endPoint: String,
+                          paramsGetter: PayloadGetterFunction? = nil) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: paramsGetter)
   }
   
   
   init(endPoint: String,
        transport: @escaping ((URL) -> WebSocketClient),
-       params: [String: Any]? = nil) {
+       params: PayloadGetterFunction? = nil) {
```

Comment:
Since this one is internal, it is safe to rename to `paramsClosure: ParamsClosure` since that is the expected default